src/asciidocParser.ts

```diff
@@ -282,4 +290,104 @@ export class AsciidocParser {
       ? vscode.workspace.rootPath
       : documentPath
   }
+
+  private async showWarningRegisterExt () {
+    if (!this.isRegisterAsciidocExtensionEnabled()) {
+      return
+    }
+    if (this.apsArbiter.getAllowScripts()) {
+      return
+    }
+    if (!await this.hasExtensionInWorkspace()) {
+      return
+    }
+    await this.apsArbiter.showDialogTrustAutherOnlyOnce()
+  }
+
+  private async registerExt (registry, apsArbiter: AsciidocParserSecurityPolicyArbiter) {
+    const useKroki = vscode.workspace.getConfiguration('asciidoc', null).get('use_kroki')
+    if (useKroki) {
+      const kroki = require('asciidoctor-kroki')
+      kroki.register(registry)
+    }
+    await this.registerExtensionInWorkspace(registry, apsArbiter)
+  }
+
+  public async hasExtensionInWorkspace () {
+    const files = await this.getExtensionFilesInWorkspace()
+    return files.length !== 0
+  }
+
+  private async readdirsRecursive (uri: vscode.Uri): Promise<[ string, vscode.FileType ][]> {
+    if (!vscode.workspace.fs.stat(uri)) {
+      return []
+    }
+    const rd = await vscode.workspace.fs.readDirectory(uri)
+    const result = rd.filter(function (extfile) {
+      return extfile[1] === vscode.FileType.File
+    })
+    for (const fd of rd) {
+      const fname = fd[0]
+      const ftype = fd[1]
+      if (ftype === vscode.FileType.Directory) {
+        const subdir = await this.readdirsRecursive(uri.with({ path: path.posix.join(uri.path, fname) }))
+        for (const subdirfile of subdir) {
+          result.push([path.posix.join(fname, subdirfile[0]), subdirfile[1]])
+        }
+      }
+    }
+    return result
+  }
+
+  private async getExtensionFilesInWorkspace (): Promise<[ string, vscode.FileType ][]> {
+    const workspaceFolders = vscode.workspace.workspaceFolders
+    if (workspaceFolders === undefined) {
+      return []
+    }
+    const workspacePath = workspaceFolders[0].uri
+    const extDir = path.posix.join(workspacePath.path, extDirInWorkspace)
+    if (!vscode.workspace.fs.stat(workspacePath.with({ path: extDir }))) {
+      return []
+    }
+    const rd = await this.readdirsRecursive(workspacePath.with({ path: extDir }))
+    return rd.filter(function (extfile) {
+      return extfile[1] === vscode.FileType.File && extfile[0].endsWith('.js')
+    })
+  }
```

Comment:
I think we should use `vscode.workspace.findFiles`: https://code.visualstudio.com/api/references/vscode-api#workspace

> Find files across all [workspace folders](https://code.visualstudio.com/api/references/vscode-api#workspace.workspaceFolders) in the workspace.
> example - `findFiles('*/.js', '/node_modules/', 10)`